assets/model_monitoring/components/src/model_monitor_create_manifest/run.py

```diff
@@ -27,10 +27,19 @@ def _generate_manifest(root_dir: str):
 
 def run():
     """Create Manifest."""
-    # init aml OBO class for supporting credential-less datastore scenarios.
-    # this class will init specific env variables required by Amlfs to get
-    # user token in case of credential-less data access scenario
-    AzureMLOnBehalfOfCredential()
+    # TODO: investigate why these aren't initialized by AzureML Filesystem itself
+    # init env-vars for supporting credential-less datastore scenarios.
+    # this env variable is required by Amlfs to get user token.
+    spark = init_spark()
+    spark_conf = spark.sparkContext.getConf()
+    spark_conf_vars = {
+        "AZUREML_SYNAPSE_CLUSTER_IDENTIFIER": "spark.synapse.clusteridentifier",
```

Comment:
I suppose these are only place holder values? They must be real cluster Id and endpoint url to take effect? If so, why we need to add them here?

Reply:
No not place holder. We get the real values from the spark conf with the key on right-side and set the environment variable with key on left-side to the spark conf value.